src/modules/routing/services/MediatorService.ts

```diff
@@ -0,0 +1,241 @@
+    }
+    for (const update of message.updates) {
+      const update_ = new KeylistUpdated({
+        action: update.action,
+        recipientKey: update.recipientKey,
+        result: KeylistUpdateResult.NoChange,
+      })
+      if (update.action === KeylistUpdateAction.add) {
+        update_.result = await this.saveRoute(update.recipientKey, mediationRecord)
+        keylist.push(update_)
+      } else if (update.action === KeylistUpdateAction.remove) {
+        update_.result = await this.removeRoute(update.recipientKey, mediationRecord)
+        keylist.push(update_)
+      }
+    }
+    // emit event to send message that notifies recipient
+    const responseMessage = new KeylistUpdateResponseMessage({ keylist })
+    const outbound = createOutboundMessage(connection, responseMessage)
+    if (message.hasReturnRouting()) {
+      return outbound
+    } else {
+      await this.messageSender.sendMessage(outbound)
+    }
+  }
+
+  public async saveRoute(recipientKey: Verkey, mediationRecord: MediationRecord | null): Promise<KeylistUpdateResult> {
+    if (mediationRecord) {
+      mediationRecord.recipientKeys.push(recipientKey)
+      this.mediationRepository.update(mediationRecord)
+      return KeylistUpdateResult.Success
+    }
+    return KeylistUpdateResult.ServerError
+  }
+
+  public async removeRoute(recipientKey: Verkey, mediationRecord: MediationRecord | null) {
+    if (mediationRecord) {
+      const index = mediationRecord.recipientKeys.indexOf(recipientKey, 0)
+      if (index > -1) {
+        mediationRecord.recipientKeys.splice(index, 1)
+      }
+      this.mediationRepository.update(mediationRecord)
+      return KeylistUpdateResult.Success
+    }
+    return KeylistUpdateResult.ServerError
+  }
+
+  public async findRecipient(recipientKey: Verkey): Promise<string | null> {
+    const records = await this.getAll()
+    for (const record of records) {
+      for (const key of record.recipientKeys) {
+        if (recipientKey == key) {
+          return record.connectionId
+        }
+      }
+    }
+    return null
+  }
+  // TODO: remove this in favor of find by connection id
+  public async findRecipientByConnectionId(connectionId: string): Promise<MediationRecord | null> {
+    try {
+      const records = await this.mediationRepository.findByQuery({ connectionId })
+      return records[0]
+    } catch (error) {
+      return null
+    }
+  }
+  // TODO: resolve possible duplicate keylist messages
+  //public async createKeylistUpdateResponseMessage(keylist: KeylistUpdated[]): Promise<KeylistUpdateResponseMessage> {
+  //const keylistUpdateMessage = new KeylistUpdateResponseMessage({
+  //  updated: keylist,
+  //})
+  //return keylistUpdateMessage
+  //}
+
+  public async createGrantMediationMessage(mediation: MediationRecord): Promise<MediationGrantMessage> {
+    if (this.routingKeys.length === 0) {
+      /* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
```suggestion
      /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```
I totally overlook that one is `eslint-disable` and other one is `eslint-enable` 😄